libs/domains/service-helm/feature/src/lib/values-override-files-setting/values-override-files-setting.tsx

```diff
@@ -0,0 +1,170 @@
+import { type GitProviderEnum, type HelmRequestAllOfSource } from 'qovery-typescript-axios'
+import { type PropsWithChildren, type ReactNode, useState } from 'react'
+import { Controller, type UseFormReturn } from 'react-hook-form'
+import { useParams } from 'react-router-dom'
+import { PREVIEW_CODE } from '@qovery/shared/routes'
+import { Button, Heading, Icon, IconAwesomeEnum, InputSelect, Popover, Section } from '@qovery/shared/ui'
+import useHelmDefaultValues from '../hooks/use-helm-default-values/use-helm-default-values'
+import ValuesOverrideYamlSetting from '../values-override-yaml-setting/values-override-yaml-setting'
+
+export type ValuesOverrideTypes = 'GIT_REPOSITORY' | 'YAML' | 'NONE'
+
+export interface HelmValuesFileData {
+  type: ValuesOverrideTypes
+  repository?: string
+  provider?: GitProviderEnum
+  branch?: string
+  paths?: string
+  content?: string
+}
+
+export interface ValuesOverrideFilesSettingProps {
+  source: HelmRequestAllOfSource
+  methods: UseFormReturn<HelmValuesFileData>
+  watchFieldType: ValuesOverrideTypes
+  gitRepositorySettings: ReactNode
+  onSubmit: () => void
+}
+
+export function ValuesOverrideFilesSetting({
+  methods,
+  source,
+  watchFieldType,
+  children,
+  gitRepositorySettings,
+  onSubmit,
+}: PropsWithChildren<ValuesOverrideFilesSettingProps>) {
+  const { environmentId = '' } = useParams()
+
+  const [enabledHelmDefaultValues, setEnabledHelmDefaultValues] = useState(false)
+
+  const { refetch: refetchHelmDefaultValues, isFetching: isLoadingHelmDefaultValues } = useHelmDefaultValues({
+    environmentId,
+    helmDefaultValuesRequest: {
+      source,
+    },
+    enabled: enabledHelmDefaultValues,
+  })
+
+  const createHelmDefaultValuesMutation = async () => {
+    setEnabledHelmDefaultValues(true)
+    const { data: helmDefaultValues } = await refetchHelmDefaultValues()
```

Comment:
IMO fetching helm default values everytime even when user doesn't click on the button is simpler and the request cost is not very high.
Here looks like you are fighting against react-query principles.
If that's the behavior that you want we can maybe have Helm default values as both query AND mutation for flexible usage.

Reply:
I don't want to fetch before because we have an loading on the button, I think it more smooth to keep it and in the spec they mentions to load on the click only

It's inspired by this answer of Tkdodo : https://stackoverflow.com/a/66891350, he proposes two solutions and the second with mutation but I prefer the first to avoid duplicate requests and because we use this behavior only one time, but I can do that as well